tidb-server/server/server_test.go

```diff
@@ -220,6 +221,26 @@ func runTestConcurrentUpdate(c *C) {
 	})
 }
 
+func runTestErrorCode(c *C) {
+	runTests(c, dsn, func(dbt *DBTest) {
+		dbt.mustExec("create table test (c int PRIMARY KEY);")
+		dbt.mustExec("insert into test values (1);")
+		txn1, err := dbt.db.Begin()
+		c.Assert(err, IsNil)
+		_, err = txn1.Exec("insert into test values(1)")
+		c.Assert(err, IsNil)
+		err = txn1.Commit()
+		checkErrorCode(c, err, tmysql.ErrDupEntry)
+	})
+}
+
+func checkErrorCode(c *C, e error, code uint16) {
+	c.Assert(e, NotNil)
```

Comment:
no need to check not nil here, if e is nil, following type conversation is not ok.